src/providers/postgres/qgspostgresexpressioncompiler.cpp

```diff
@@ -32,3 +38,150 @@ QString QgsPostgresExpressionCompiler::quotedValue( const QVariant& value, bool&
   return QgsPostgresConn::quotedValue( value );
 }
 
+static const QMap<QString, QString>& functionNamesSqlFunctionsMap()
+{
+  static QMap<QString, QString> fnNames;
+  if ( fnNames.isEmpty() )
+  {
+    fnNames =
+    {
+      { "sqrt", "sqrt" },
+      { "radians", "radians" },
+      { "degrees", "degrees" },
+      { "abs", "abs" },
+      { "cos", "cos" },
+      { "sin", "sin" },
+      { "tan", "tan" },
+      { "acos", "acos" },
+      { "asin", "asin" },
+      { "atan", "atan" },
+      { "atan2", "atan2" },
+      { "exp", "exp" },
+      { "ln", "ln" },
+      { "log", "log" },
+      { "log10", "log10" },
+      { "round", "round" },
+      { "floor", "floor" },
+      { "ceil", "ceil" },
+      { "pi", "pi" },
+      // geometry functions
+      { "azimuth", "ST_Azimuth" },
+      { "x", "ST_X" },
```

Comment:
Don't forget that x/y work with non points in QGIS expressions too (they return centroid x/y). This will need to be handled explicitly eg by wrapping st_x(st_centroid(...